roles/register_hypervisor/files/add_host_sdk4

```diff
@@ -0,0 +1,116 @@
+#! /usr/bin/env python
+
+import time
+
+import ovirtsdk4 as sdk
+import ovirtsdk4.types as types
+
+
+def debug(f):
+    def inner(*args, **kwargs):
+        import ipdb
+        with ipdb.launch_ipdb_on_exception():
+            return f(*args, **kwargs)
+    return inner
+
+
+# @debug
+def main():
+    args = parse_args()
+    con = get_connection(
+            url=args.url,
+            user=args.username,
+            password=args.password,
+            insecure=True
+    )
+    hosts_service = con.system_service().hosts_service()
+
+    host = add_host(hosts_service, args.host_name, args.host, args.cluster,
+                    args.host_password, args.deploy_hosted_engine)
+    wait_for_vdsm_host_up(hosts_service, host)
+
+    # Close the connection to the server:
+    con.close()
+
+
+def parse_args():
+    import argparse
+
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('--url', dest='url', required=True,
+                        help='the url for the Engine api')
+    parser.add_argument('--username', dest='username', required=True,
+                        help='the username for the Engine api')
+    parser.add_argument('--password', dest='password', required=True,
+                        help='the password for the Engine api')
+    parser.add_argument('--host', dest='host', required=True,
+                        help='the ip address or fqdn of the hypervisor host')
+    parser.add_argument('--host_name', dest='host_name', required=False,
+                        help='the host logic name')
+    parser.add_argument('--host-password', dest='host_password', required=True,
+                        help='the root password for the hypervisor')
+    parser.add_argument('--cluster', dest='cluster', required=True,
+                        help='the name of the cluster to add the hypervisor to')
+    parser.add_argument('--deploy_hosted_engine', dest='deploy_hosted_engine',
```

Comment:
Also should probably use `action='store_true'` which turns it to a flag (`--deploy-hosted-engine` would set it to True)